internal/ostreeclient/ostreeclient.go

```diff
@@ -15,41 +16,60 @@ type IClient interface {
 	Undeploy(ostreeIndex int) error
 	SetDefaultDeployment(index int) error
 	IsOstreeAdminSetDefaultFeatureEnabled() bool
+	GetDeployment(osname string) (string, error)
 }
 
 type Client struct {
 	executor ops.Execute
+	ibi      bool
 }
 
-func NewClient(executor ops.Execute) IClient {
+func NewClient(executor ops.Execute, ibi bool) IClient {
 	return &Client{
 		executor: executor,
+		ibi:      true,
 	}
 }
 
 func (c *Client) PullLocal(repoPath string) error {
-	_, err := c.executor.Execute("ostree", "pull-local", repoPath)
+	args := []string{"pull-local"}
+	if c.ibi {
+		args = append(args, "--repo", "/mnt/ostree/repo")
+	}
+	_, err := c.executor.Execute("ostree", append(args, repoPath)...)
 	return err
 }
 
 func (c *Client) OSInit(osname string) error {
-	_, err := c.executor.Execute("ostree", "admin", "os-init", osname)
+	args := []string{"admin", "os-init"}
+	if c.ibi {
+		args = append(args, "--sysroot", "/mnt")
+	}
+
+	_, err := c.executor.Execute("ostree", append(args, osname)...)
 	return err
 }
 
 func (c *Client) Deploy(osname, refsepc string, kargs []string) error {
 	args := []string{"admin", "deploy", "--os", osname}
+	if c.ibi {
+		args = append(args, "--sysroot", "/mnt")
+	}
 	args = append(args, kargs...)
 	args = append(args, refsepc)
-	if c.IsOstreeAdminSetDefaultFeatureEnabled() {
+	if c.IsOstreeAdminSetDefaultFeatureEnabled() && !c.ibi {
```

Comment:
Why not just set it as default in case of IBI?

Reply:
I will change places between them but i don't fully understand the comment